lightning/src/util/transaction_utils.rs

```diff
@@ -232,28 +226,30 @@ mod tests {
 		// Check that we never add dust outputs
 		let mut tx = Transaction { version: 2, lock_time: 0, input: Vec::new(), output: Vec::new() };
 		let orig_wtxid = tx.wtxid();
+		let output_spk = Script::new_p2pkh(&PubkeyHash::hash(&[0; 0]));
+		assert_eq!(output_spk.dust_value().as_sat(), 546);
 		// 9 sats isn't enough to pay fee on a dummy transaction...
 		assert_eq!(tx.get_weight() as u64, 40); // ie 10 vbytes
-		assert!(maybe_add_change_output(&mut tx, 9, 0, 253, Script::new()).is_err());
+		assert!(maybe_add_change_output(&mut tx, 9, 0, 250, output_spk.clone()).is_err());
```

Comment:
I'm confused why the values used changed in this test? I'd expect them to stay the same after switching to rust-bitcoin's `dust_value`

Reply:
Because our existing code is wrong for all values except one or two script types (of which empty-script is not one).